source/github-helpers/get-default-branch.ts

```diff
@@ -31,16 +31,26 @@ const _getDefaultBranch = cache.function('default-branch', async (repository: pa
 	`);
 
 	return response.repository.defaultBranchRef.name;
-}, {
-	maxAge: {hours: 1},
-	staleWhileRevalidate: {days: 20},
-	cacheKey: ([repository]) => repository.nameWithOwner,
-});
-
-export default async function getDefaultBranch(repository: pageDetect.RepositoryInfo | undefined = getRepo()): Promise<string> {
-	if (!repository) {
-		throw new Error('getDefaultBranch was called on a non-repository page');
-	}
+}
+
+// DO NOT use optional arguments/defaults in "cached functions" because they can't be memoized effectively
+// https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1864
+export const getDefaultBranchOfRepo = cache.function('default-branch',
+	async (repository: RepositoryInfo): Promise<string> => {
```

Comment:
The name clarifies that it refers to a specific repo, not the current repo/page.